maven-plugin/src/main/java/io/confluent/kafka/schemaregistry/maven/derive/schema/DeriveAvroSchema.java

```diff
@@ -0,0 +1,268 @@
+  }
+
+  /**
+   * Using mergeRecords to combine number and union types of fields
+   */
+  protected ArrayNode mergeMultipleMessages(List<JsonNode> uniqueSchemas,
+                                            Map<JsonNode, ArrayNode> schemaToIndex) {
+
+    // Using index to identify uniqueSchemas rather than map
+    // mergeRecords() can change schema such that it doesn't match any original schema
+    ArrayNode schemaInfoList = mapper.createArrayNode();
+    List<ArrayNode> originalSchemaMatches = new ArrayList<>();
+    uniqueSchemas.forEach(o -> originalSchemaMatches.add(schemaToIndex.get(o)));
+
+    // Merge schema with same name and different data types and edit schemas in-place
+    try {
+      mergeRecords(uniqueSchemas);
+    } catch (IllegalArgumentException ignored) {
+      // Ignoring if there are any conflicts in merging
+    }
+
+    // Find unique schema after merging records and collect which messages it matches
+    Map<JsonNode, ArrayNode> schemaToMatches = new HashMap<>();
+    uniqueSchemas.forEach(s -> schemaToMatches.put(s, mapper.createArrayNode()));
+    for (int i = 0; i < uniqueSchemas.size(); i++) {
+      schemaToMatches.get(uniqueSchemas.get(i)).addAll(originalSchemaMatches.get(i));
+    }
+    for (JsonNode uniqueSchema : DeriveSchemaUtils.getUnique(uniqueSchemas)) {
+      updateSchemaInformation(uniqueSchema, schemaToMatches.get(uniqueSchema), schemaInfoList);
+    }
+    return schemaInfoList;
+  }
+
+  @Override
+  protected ObjectNode mergeMultipleDataTypes(ObjectNode mergedArray,
+                                              List<JsonNode> primitives,
+                                              List<JsonNode> records,
+                                              List<JsonNode> arrays,
+                                              boolean check2dArray) {
+    mergeUnions(records, primitives);
+    DeriveSchemaUtils.mergeNumberTypes(primitives);
+    primitives = DeriveSchemaUtils.getUnique(primitives);
+    List<JsonNode> dataTypes = new ArrayList<>();
+    // To recursively merge number types in-place for records and arrays, the result is ignored
+    if (records.size() > 0) {
+      mergeRecords(records);
+    }
+    if (arrays.size() > 0) {
+      dataTypes.add(mergeArrays(arrays, true, false));
+    }
+    for (List<JsonNode> types : Arrays.asList(records, primitives)) {
+      dataTypes.addAll(types);
+    }
+    ArrayNode items = mapper.createArrayNode().addAll(DeriveSchemaUtils.getUnique(dataTypes));
+    getSingleDataType(mergedArray, items);
+    return mergedArray;
+  }
+
+  /**
+   * Convert json schema template to avro format and validate generated schema
+   */
+  protected JsonNode convertToFormat(JsonNode schema, String name) {
+    ObjectNode schemaForRecord = convertToFormatForRecord(schema, name);
+    AvroSchema avroSchema = new AvroSchema(schemaForRecord.toString());
+    avroSchema.validate();
```

Comment:
How are we checking the outcome of this validation? If its an exception being thrown, we aren't catching it. If there is a return status code, we aren't checking.